flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClientTest.java

```diff
@@ -456,6 +461,20 @@ public void testCheckAndUpdateConfigMapWhenReplaceConfigMapFailed() throws Excep
         }
     }
 
+    @Test
+    public void testWatchConfigMaps() throws Exception {
+        flinkKubeClient.watchConfigMaps(TESTING_CONFIG_MAP_NAME, new NoOpWatchCallbackHandler<>());
```

Comment:
Is it possible to use the `DefaultKubeClientFactory` here to construct a `KubeClient` which talks to the testing server? Otherwise we are testing here that the test setup initializes the testing `flinkKubeClient` with a namespace. To verify this, you could simply remove `config.setNamespace(namespace);` from `DefaultKubeClientFactory` and the test should still pass.

Reply:
I also have this concern. And that is why I did not add the unit test at the very begging. Making `KubernetesMockServer` work with a real `KubeClient` is not very easy. I will try to dig more.

If we could not git it done, I will add an integration test using `minikube` and a real `KubeClient`.

Reply:
Why is it difficult to connect to the `KubernetesMockServer`? If this should be too complicated, then we can also leave it out. I just assumed that this would be somehow possible.

Reply:
I used to believe that generating a kube config file for `MockKubernetesServer` is a little difficult. Because we need to generate and set proper certificate authority file. But after more investigation, I find that we could easily skip tls verification. So I will update the test to make it more appropriate, which should fail when the `config.setNamespace(namespace)` is not called.

Reply:
Great :-)